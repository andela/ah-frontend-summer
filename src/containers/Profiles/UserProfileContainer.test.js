import React from 'react';
import { shallow } from 'enzyme';
import { 
    UserProfileContainer, 
    mapDispatchToProps 
} from './UserProfileContainer';

describe("UserProfile", () => {
    const fetchUserProfileMock = jest.fn();
    const fetchUserArticlesMock = jest.fn();
    const userProfileProps = {
        fetchUserProfile: fetchUserProfileMock,
        fetchUserArticles: fetchUserArticlesMock,
        match: { params: { username: "" }},
        profile: {},
        dismissMessage: jest.fn()
    };
    const userProfileContainer = shallow(<UserProfileContainer {...userProfileProps}/>);
    const userProfileContainerInstance = userProfileContainer.instance();

    it("renders correctly", () => {
        expect(userProfileContainer).toMatchSnapshot();
    });
    
    it("componentDidMount calls dispatches actions to fetch user profile and user articles", () => {
        userProfileContainerInstance.componentDidMount();
        expect(fetchUserProfileMock).toHaveBeenCalled();
        expect(fetchUserArticlesMock).toHaveBeenCalled();
    });

    it("mapDispatchToProps", () => {
        const dispatch = jest.fn();
        mapDispatchToProps(dispatch).fetchUserProfile();
        mapDispatchToProps(dispatch).fetchUserArticles();
        mapDispatchToProps(dispatch).dismissMessage();
        expect(dispatch.mock.calls.length).toBe(3);
    });
});
