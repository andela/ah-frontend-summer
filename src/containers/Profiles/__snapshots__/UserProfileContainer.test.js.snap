// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile renders correctly 1`] = `
<Connect(UserProfile)
  dismissMessage={[MockFunction]}
  fetchUserArticles={
    [MockFunction] {
      "calls": Array [
        Array [
          "https://ah-backend-summer-staging.herokuapp.com/api/v1/articles/?author=",
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
  fetchUserProfile={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "token": null,
            "username": "",
          },
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
  match={
    Object {
      "params": Object {
        "username": "",
      },
    }
  }
  onDismissHandler={[Function]}
  profile={Object {}}
/>
`;
